vars/filebeat.groovy

```diff
@@ -78,14 +84,29 @@ def start(Map args = [:]) {
     output: output,
     config: config,
     image: image,
-    workdir: workdir,
     timeout: timeout,
     archiveOnlyOnFail: archiveOnlyOnFail
   ]
 
   writeJSON(file: "${workdir}/filebeat_container_${env.NODE_NAME}.json", json: json)
 }
 
+/**
+* This method:
+*  1. removes leading and trailing whitespaces, using trim()
+*  2. replaces inner whitespaces with '_', using a regex
+*  3. replaces any occurrences of non-characters or numbers or dots, with a '_', using a regex
+*
+* For simplicity, it could lead to collitions for different inputs: as an example,
+* both "a&b" and "a|b" will result in "a_b".
+* 
+* @param input the string to be sanitised
+* @return the sanitised ouput
+*/
+def String sanitizeOutputFileName(String input){
+  return input?.trim().replaceAll("\\s","_").replaceAll("[^a-zA-Z0-9.]+", "_")
+}
+
 def stop(Map args = [:]){
   def workdir = args.containsKey('workdir') ? args.workdir : pwd()
```

Comment:
the parameter is removed see https://github.com/elastic/apm-pipeline-library/pull/1027/files#r594183293